packages/core/src/scanning.ts

```diff
@@ -109,3 +109,27 @@ export const scanOutputs = (
 
     return matches;
 };
+
+export const scanOutputsWithTweak = (
+    spendPublicKey: Buffer,
+    scanTweak: Buffer,
+    outputs: Buffer[],
+): Map<string, Buffer> => {
+    const matches = new Map<string, Buffer>();
+
+    const tweakedPublicKey = secp256k1.publicKeyTweakAdd(
+        spendPublicKey,
+        scanTweak,
+        true
+    );
+
+    for (let i = outputs.length - 1; i >= 0; i--) {
+        const output = outputs[i];
+        if (output.equals(tweakedPublicKey)) {
+            matches.set(output.toString('hex'), scanTweak);
+            outputs.splice(i, 1);
+        }
+    }
+
+    return matches;
+};
```

Comment:
I crucially believe this is not needed, there exists a function already with the signature 
``` typescript
export const scanOutputs = (
    scanPrivateKey: Buffer,
    spendPublicKey: Buffer,
    sumOfInputPublicKeys: Buffer,
    inputHash: Buffer,
    outputs: Buffer[],
    labels?: LabelMap,
): Map<string, Buffer>
```

Reply:
We should take advantage of the information available in the silent block. @theanmolsharma has approved the creation of a separate function for this purpose.

Reply:
@notTanveer,  Could you clarify what information you need, that we are not making use of  in the existing function ?